src/components/AssetsPage/index.jsx

```diff
@@ -1,41 +1,34 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
-import WrappedAssetsDropZone from '../AssetsDropZone';
-import WrappedAssetsTable from '../AssetsTable';
-import WrappedAssetsFilters from '../AssetsFilters';
-import WrappedPagination from '../Pagination';
 
-import { getAssets } from '../../data/actions/assets';
+import { assetActions } from '../../data/constants/actionTypes';
+import { getAssets, clearFilters } from '../../data/actions/assets';
+import { hasSelectedFilters } from '../../utils/getAssetsFilters';
+import * as pageTypes from './pageTypes';
 import edxBootstrap from '../../SFE.scss';
 import styles from './AssetsPage.scss';
 
 export class AssetsPage extends React.Component {
   constructor(props) {
     super(props);
 
-    const hasAssets = this.hasAssets(props.assetsList);
-
     this.state = {
-      checked: false,
-      hasAssets,
+      pageType: pageTypes.types.SKELETON,
```

Comment:
The initial page type is the SKELETON page. The page type changes as soon as the component receives new props (e.g. a new assets list). This is why, in the tests, I have to always call setProps to trigger a page change. If a page is initialized with data, it will still be the SKELETON page. The reason for this is what I can't call the getPageType function from the constructor, since it relies on reading the state. Is this okay with you guys?

Reply:
If we ever added, for example, server-side rendering, we might have to change this because then the page would actually be rendered once with all of the data and no ajax calls. But, since that's not planned I think this is okay.

Reply:
Okay, sounds good!